src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_tr.properties

```diff
@@ -28,7 +28,7 @@ illegal.token.text=İzin verilmeyen bir kalıp kullanılmış ''{0}''.
 illegal.type=''{0}'' tipinin kullanılmasına izin verilmez.
 inline.conditional.avoid=Satır içi koşullu ifadeler kullanılmamalıdır.
 instantiation.avoid={0} sınıfının yeni nesnesi üretilmemelidir.
-magic.number=''{0}'' sayısı ''magic number''dır. \u00D6nce tanımlanmalı, sonra kullanılmalı.
+magic.number=''{0}'' sayısı ''magic number''dır. Önce tanımlanmalı, sonra kullanılmalı.
```

Comment:
@romani There is nothing wrong with this right?

Reply:
\u00D6 is the escape sequence for Ö. This was the sole escape sequence in the entire translations and even Chinese and Japanese aren't escaped (and would be unintelligible for humans if they were). Also, there is a bunch of other Ö's there, but only this one was escaped somehow. So, I saw no point in having just this being escaped.

Reply:
There is an issue with .properties files encoding in Java:
https://stackoverflow.com/questions/59479438/really-true-that-properties-files-must-be-encoded-in-iso-8859-1-before-java-9

Reply:
@pbludov I assume this only matters for the JRE running checkstyle and not version it is compiled with (IE: If someone runs with JRE 9 though we are 8, it won't be an issue for them)?
Is there a way we can test if property files are in the correct encoding since we still work with Java 8?

Reply:
Just informational from what I found online on how to check. Thinking may be a separate issue if it is deemed.
https://unix.stackexchange.com/questions/11602/how-can-i-test-the-encoding-of-a-text-file-is-it-valid-and-what-is-it
````
/checkstyle/src/main/resources/com/puppycrawl/tools/checkstyle/checks$ file -i *
messages_de.properties: text/plain; charset=utf-8
messages_es.properties: text/plain; charset=utf-8
messages_fi.properties: text/plain; charset=utf-8
messages_fr.properties: text/plain; charset=utf-8
messages_ja.properties: text/plain; charset=utf-8
messages.properties:    text/plain; charset=us-ascii
messages_pt.properties: text/plain; charset=utf-8
messages_tr.properties: text/plain; charset=utf-8
messages_zh.properties: text/plain; charset=utf-8
````

Reply:
The `Violation.Utf8Control` class ensures that the property files are read as UTF-8 even in Java 8. It specifically creates a `Reader` for UTF-8 and uses it to open. The relevant part of the code is this:

```java
                    try (Reader streamReader = new InputStreamReader(connection.getInputStream(),
                            StandardCharsets.UTF_8.name())) {
                        // Only this line is changed to make it read property files as UTF-8.
                        resourceBundle = new PropertyResourceBundle(streamReader);
                    }
```

[Java 8 `PropertyResourceBundle` javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/PropertyResourceBundle.html) states that:

> **Note:** `PropertyResourceBundle` can be constructed either from an `InputStream` or a `Reader`, which represents a property file. Constructing a `PropertyResourceBundle` instance from an `InputStream` requires that the input stream be encoded in ISO-8859-1. In that case, characters that cannot be represented in ISO-8859-1 encoding must be represented by Unicode Escapes as defined in section 3.3 of _The Java™ Language Specification_ whereas the other constructor which takes a Reader does not have that limitation.

So, I think that we are OK here even in Java 8. In fact, it seems that someone actively took care of that in the past.

Reply:
we have this escapes from the begning
https://github.com/checkstyle/checkstyle/blame/a8b8375b1db65b131938111ff283baed771161ed/src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_tr.properties

, there might be some very old reason for it. But we switched to UTF8 few years ago
https://github.com/checkstyle/checkstyle/commit/dcb3029c008907edf6aa4b59a51349f0ca94f54b
so all that problems might be not relevant any more.